File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        p1 = 0
        p2 = 0
        res = ""
        for _ in range(len(word1) + len(word2)):
            if p1 < len(word1):
                res += word1[p1]
                p1 += 1
            if p2 < len(word2):
                res += word2[p2]
                p2 += 1
        return res

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String w1, String w2) {
        StringBuilder sb = new StringBuilder();
        int p1 = 0;
        int p2 = 0;
        while (p1 < w1.length() || p2 < w2.length()) {
            if (p1 < w1.length()) {
                sb.append(w1.charAt(p1));
                p1++;
            }
            if (p2 < w2.length()) {
                sb.append(w2.charAt(p2));
                p2++;
            }
        }
        return sb.toString();
    }
}